.github/workflows/commitlint.yml

```
@@ -0,0 +1,42 @@
+name: Commitlint
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - edited
+  merge_group:
+    branches:
+      - main
+    types:
+      - checks_requested
```

One last thing, please specify permissions.

```suggestion
      - checks_requested
permissions:
  - contents: read
```